src/components/Header/Header.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { postTodo } from '../../api/todos';
+import { formatTodo } from '../../utils/formatResponse';
+
+type Props = {
+  todosToRender: Todo[];
+  setTodosToRender: (arr: Todo[]) => void;
+  pushError: (title: string) => void;
+  setTempTodo: (todo: Todo | null) => void;
+  setIsTempLoading: (val: boolean) => void;
+  setTodos: (arr: Todo[]) => void;
+};
+
+export const Header: React.FC<Props> = ({
+  todosToRender,
+  setTodosToRender,
+  pushError,
+  setTempTodo,
+  setIsTempLoading,
+  setTodos,
+}) => {
+  const [value, setValue] = useState('');
+  const [isDisabled, setIsDisabled] = useState(false);
+  const [focus, setFocus] = useState(false);
+  const refInput = useRef<HTMLInputElement>(null);
+
+  const onInputChange = (str: string) => {
+    setValue(() => str);
+  };
+
+  // this would be handled using methods later
+  const toggleAll = () => {
+    if (todosToRender) {
+      setTodosToRender([...todosToRender].map(todo => {
+        return todosToRender.every(item => item.completed)
+          ? {
+            ...todo,
+            completed: !todo.completed,
+          }
+          : {
+            ...todo,
+            completed: true,
+          };
+      }));
+    }
+  };
+
+  const onFormSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    if (!value) {
+      pushError('Title can\'t be empty');
+
+      return;
+    }
+
+    const rawTodo = {
+      title: value,
```

Comment:
Avoid using hardcoded 'userId'. Instead, consider fetching the user's ID dynamically or passing it down as a prop.